### PR TITLE
#262: Full Flyway integration

### LINKED ISSUE #262: Full Flyway integration

### PR DESCRIPTION
**Pull request by benedeki**
* numbered the SQL scripts for Flyway
* changed `PartitioningForDB` class to contain `keysToValuesMap` instead of `keysToValues` according to validation
* added sbt alias `testAll` for running all tests
* database test for validation function moved to proper folder
* set `flywayBaselineVersion` for upgrade purposes

While it's almost 70 files, changes are minimal. In majority of cases it's just a rename.

---
For the actual upgrade deployment these steps are necessary:
#### Manual:
`sbt -Dflyway.url=<target db url> -Dflyway.password=<postgres user password> flywayBaseline`

#### Add to deployment automation
`sbt -Dflyway.url=<target db url> -Dflyway.password=<postgres user password> flywayMigrate`

where
* `<target db url>` is the database towards which the deployment is running, e.g. _"jdbc:postgresql://localhost:5432/atum_db"_
* `<postgres user password>` is the password for the _postgres_ super user at the above database
---

Depends on #265 
Depends on #267 
Depends on #272 
Depends on #275 

Closes #262

### CODE REVIEW

**Comment by benedeki** [comment]
Release notes:
Reordered and fixed database structure scripts to allow Flyway usage for incremental migration of existing database (version 0.1.x). Known issues: there might be needed some manual steps if upgrading from database of version 0.2.0.

**Comment by benedeki** [comment]
This item depends on:
* AbsaOSS/atum-service#265
* AbsaOSS/atum-service#267
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.

**Comment by benedeki** [comment]
This item depends on:
* ~~AbsaOSS/atum-service#265~~
* ~~AbsaOSS/atum-service#267~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.

**Comment by benedeki** [comment]
This item depends on:
* ~~AbsaOSS/atum-service#265~~
* ~~AbsaOSS/atum-service#267~~
* ~~AbsaOSS/atum-service#272~~
* ~~AbsaOSS/atum-service#275~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.

File: database/src/main/postgres/public/V0.1.0.11__global_id.sql [hunk 1]
```diff
@@ -26,7 +26,7 @@ CREATE SEQUENCE IF NOT EXISTS public.global_id_seq
 
 CREATE OR REPLACE FUNCTION public.global_id() RETURNS BIGINT AS
 $$
--------------------------------------------------------------------------------
+    -------------------------------------------------------------------------------
```
Comment (lsulak) [inline database/src/main/postgres/public/V0.1.0.11__global_id.sql:29]: not needed

File: database/src/main/postgres/validation/V0.2.0.6__validate_partitioning.sql [hunk 2]
```diff
@@ -15,26 +15,27 @@
...
 CREATE OR REPLACE FUNCTION validation.validate_partitioning(
-    IN i_partitioning JSONB,
-    IN i_strict_check BOOLEAN = true
+    IN  i_partitioning JSONB,
+    IN  i_strict_check BOOLEAN = true,
+    OUT error_message  TEXT
 ) RETURNS SETOF TEXT AS
 $$
--------------------------------------------------------------------------------
+    -------------------------------------------------------------------------------
```
Comment (lsulak) [inline database/src/main/postgres/validation/V0.2.0.6__validate_partitioning.sql:23]: not needed

File: database/src/main/postgres/validation/V0.2.0.7__is_partitioning_valid.sql [hunk 3]
```diff
@@ -15,11 +15,12 @@
...
 CREATE OR REPLACE FUNCTION validation.is_partitioning_valid(
-    IN i_partitioning   JSONB,
-    IN i_strict_check   BOOLEAN = true
+    IN  i_partitioning   JSONB,
+    IN  i_strict_check   BOOLEAN = true,
+    OUT is_valid         BOOLEAN
 ) RETURNS BOOLEAN AS
 $$
--------------------------------------------------------------------------------
+    -------------------------------------------------------------------------------
```
Comment (lsulak) [inline database/src/main/postgres/validation/V0.2.0.7__is_partitioning_valid.sql:23]: not needed

**Review by lsulak — COMMENTED** [review]
LGTM, found a few minor things. But I want to ask: why the numbering like this? For example there is no `V0.1.0.3.{sql|ddl}` file, so there are gaps. There is `V0.1.0.30.ddl` instead. So we don't want increments?

**Comment by benedeki** [comment]
> LGTM, found a few minor things. But I want to ask: why the numbering like this? For example there is no `V0.1.0.3.{sql|ddl}` file, so there are gaps. There is `V0.1.0.30.ddl` instead. So we don't want increments?

It is incremental, but with gaps.
I put the gaps there between folders/schemas number ranges, so in case of testing, if something has to change, I won't be forced to renumber everything. And it actually happened 😉

**Pull request merged**